lambda/translmod.ml

```diff
@@ -919,40 +928,136 @@ let required_globals ~flambda body =
   Translprim.clear_used_primitives ();
   required
 
+let add_arg_block_to_module_block primary_block_lam size restr =
+  let primary_block_id = Ident.create_local "*primary-block*" in
+  let arg_block_id = Ident.create_local "*arg-block*" in
+  let arg_block_lam =
+    apply_coercion Loc_unknown Strict restr (Lvar primary_block_id)
+  in
+  let get_field i = Lprim (mod_field i, [Lvar primary_block_id], Loc_unknown) in
+  let all_fields = List.init size get_field @ [Lvar arg_block_id] in
```

Comment:
Looking at the code here, it seems that "arg_block" is a misnomer? It should be called "the field in the main block that represents -as-argument-for". I.e., it's a field in a block, not a block.

Reply:
I'm not sure what you mean: `arg_block_lam` is exactly the Lambda code that builds the argument block, `arg_block_field` is the field in which we put the argument block, and `arg_block_id` is a local identifier where we bind the argument block. It very much is a block (`apply_coercion` will return either a reference to a block or a `Pmakeblock` expression).

Reply:
As elsewhere, I think “argument block” is an important concept that should be called out

Reply:
OK then how about `arg_block_idx`? I see you've adopted `_idx` elsewhere.

Reply:
Hmm! Yeah, it's not clear that the word _field_ is doing anything there, and _idx_ is good.